plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterDestroyWorker.java

```diff
@@ -239,7 +240,10 @@ public boolean destroy() throws CloudRuntimeException {
                 }
             } else {
                 try {
-                    deleteKubernetesClusterNetworkRules();
+                    NetworkVO kubernetesClusterNetwork = networkDao.findById(kubernetesCluster.getNetworkId());
+                    if (kubernetesClusterNetwork != null && kubernetesClusterNetwork.getGuestType() != Network.GuestType.Shared) {
+                        deleteKubernetesClusterNetworkRules();
+                    }
```

Comment:
I know I'm not a nice guy, but can we please have new code in separate methods:
```
private void checkForRulesToDelete(){
    NetworkVO kubernetesClusterNetwork = networkDao.findById(kubernetesCluster.getNetworkId());
    if (kubernetesClusterNetwork != null && kubernetesClusterNetwork.getGuestType() != Network.GuestType.Shared) {
        deleteKubernetesClusterNetworkRules();
    }
}
```
and then
```suggestion
                    checkForRulesToDelete();
```